### PR TITLE
[SPARK-47599][MLLIB] MLLib: Migrate logWarn with variables to structured logging framework

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to migrate `logWarn` in module `MLLib` with variables to `structured logging framework`.

### Why are the changes needed?
To enhance Apache Spark's logging system by implementing structured logging.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
- Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by gengliangwang** [comment]
Let's continue on this one :)

**Comment by panbingkun** [comment]
> Let's continue on this one :)

Okay, let me check it first, 😄

File: mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala [hunk 1]
```diff
@@ -451,8 +451,8 @@ class KMeans @Since("1.5.0") (
 
   private def trainWithBlock(dataset: Dataset[_], instr: Instrumentation) = {
     if (dataset.storageLevel != StorageLevel.NONE) {
-      instr.logWarning(s"Input vectors will be blockified to blocks, and " +
```
Comment (panbingkun) [inline mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala:454]: Delete unnecessary string interpreter `s"..."`

File: mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala [hunk 2]
```diff
@@ -179,8 +179,8 @@ class LinearSVC @Since("2.2.0") (
       maxBlockSizeInMB)
 
     if (dataset.storageLevel != StorageLevel.NONE) {
-      instr.logWarning(s"Input instances will be standardized, blockified to blocks, and " +
```
Comment (panbingkun) [inline mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala:182]: Delete unnecessary string interpreter. `s"..."`

File: mllib/src/main/scala/org/apache/spark/ml/feature/StopWordsRemover.scala [hunk 3]
```diff
@@ -129,9 +130,9 @@ class StopWordsRemover @Since("1.5.0") (@Since("1.5.0") override val uid: String
     if (Locale.getAvailableLocales.contains(Locale.getDefault)) {
       Locale.getDefault
     } else {
-      logWarning(s"Default locale set was [${Locale.getDefault.toString}]; however, it was " +
```
Comment (panbingkun) [inline mllib/src/main/scala/org/apache/spark/ml/feature/StopWordsRemover.scala:132]: `toString` is `redundant` in our scenario. Let's remove it.

**Comment by panbingkun** [comment]
> Let's continue on this one :)

Ready for it!

File: mllib/src/main/scala/org/apache/spark/mllib/tree/model/treeEnsembleModels.scala [hunk 4]
```diff
@@ -407,15 +407,19 @@ private[tree] object TreeEnsembleModel extends Logging {
...
+            log"(or larger).")
         }
       } else {
         if (sc.executorMemory <= memThreshold) {
-          logWarning(s"$className.save() was called, but it may fail because of too little" +
-            s" executor memory (${sc.executorMemory}m)." +
-            s"  If failure occurs try setting executor-memory ${memThreshold}m (or larger).")
+          logWarning(log"${MDC(LogKeys.CLASS_NAME, className)}.save() was called, " +
+            log"but it may fail because of too little executor memory " +
+            log"(${MDC(LogKeys.EXECUTION_MEMORY_SIZE, sc.executorMemory)}m). If failure occurs, " +
```
Comment (gengliangwang) [inline mllib/src/main/scala/org/apache/spark/mllib/tree/model/treeEnsembleModels.scala:420]:
```suggestion
            log"(${MDC(LogKeys.EXECUTOR_MEMORY_SIZE, sc.executorMemory)}m). If failure occurs, " +
```

Comment (panbingkun) [reply mllib/src/main/scala/org/apache/spark/mllib/tree/model/treeEnsembleModels.scala]: Yeah, Good catch!
Thanks, updated!

**Review by gengliangwang — APPROVED** [review]

**Comment by gengliangwang** [comment]
@panbingkun Thanks for the work. Merging to master.

**Pull request closed without merging**